### DISCUSSION TITLE
Unable to tag vlan for virtual machine in a cluster with no site

### DISCUSSION THREAD

**Discussion by nomad-cyanide** [netbox-community/netbox#8852] [General]
I have just upgraded to 3.1.9 and is facing the issue of being unable to tag a vlan on a VM.

I have a virtual cluster X with no site, because it consist of 4 machines on site1 and 4 machines on site2, connected with layer2. The hosts have their site to match the site they are physically on, but the cluster is set to "none", because it cant have different site than its hosts.
Now on a VM on that cluster X, I am unable to tag a vlan in any vlan-group, even though I can select which vlan-group I want to tag from. 
Am I missing something, or have I actually come across an error?

**Comment by mtinberg** [comment]
Would making a Site Group and associating the two Sites with it, then associating the VLAN Groups or VLANs with the Site Group instead of a Site help fix this?


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison

**Reply by nomad-cyanide** [reply]
Hi
We already have a site-group containing all the relevant sites and defined the vlan-groups as scope site-group pointing to the site-group mentioned first.
The odd thing is I can specify the vlan-group, but no vlans show up.
![Screenshot 2022-03-15 at 14 40 59](https://user-images.githubusercontent.com/47887698/158390571-9350371e-eb61-4888-8893-51d6de6a5daf.png)

That VLAN-group contains 1400+ vlans.

**Reply by nomad-cyanide** [reply]
Did a little more testing. 
It seems I can do the vlan-tagging via API (using my regular old scripts), but I cannot do it via the GUI, so it looks like a GUI interface-issue.

**Reply by cstueckrath** [reply]
we have the same problem with "un-sited" clusters

**Comment by nomad-cyanide** [comment]
There has been a new development in this.
It turns out that we can do the tagging in the test-instance, we have of netbox. The test-instance is basically a copy of prod, yet it is on the prod-instance we cannot tag vlans, where we can on the test-instance.
Could there be some permission issue or a database-issue? I find it unlikely, that it is a database issue, as the prod database has been replicated to test and it works in test.
*edit* This was a wrong statement. Read further below for clarification

**Reply by candlerb** [reply]
Are they exactly the same version of Netbox?  If so, try looking in your browser console for any AJAX errors.

**Reply by nomad-cyanide** [reply]
Version was the same. Database was not. Turned out that in test there were some vlans created without sites and those could be used to tag on clusters without site. After syncing database the issue remains the same:
We cannot tag vlans on VM in clusters that have no site in the GUI. We can do it the API, not the GUI.

**Reply by cstueckrath** [reply]
we run the latest version of netbox and have VLANs with and without sites. None of them are usable in VMs.

**Reply by candlerb** [reply]
@nomad-cyanide @cstueckrath: if you can create a clear series of steps to reproduce the issue with a fresh Netbox 3.2 install, then it can be raised as a ticket.

For me: setting a virtual interface on a VM to mode "Access" and then selecting an untagged VLAN works fine; also setting the mode to "Tagged" and selecting a set of tagged VLANs works fine.

However there are lots of variables. In my case, the VLANs assigned to a particular site, and the VM cluster is assigned to the same site, and I'm not using VLAN groups.  So you'll need to be specific around the circumstances required to reproduce this.

**Reply by cstueckrath** [reply]
* Cluster siteless
* any regional scoped VLAN Group (DCIM > location, region, site, site group)
* VLANs in that group

As it is not possible to assign a cluster to a Region or a Site group without assigning it to a site, the VLANs will not be available.
If the VLAN group is not scoped, VLANs are available for siteless clusters

Expected behaviour: Cluster Region and Site group are actually used and saved, even when no Site will be selected

**Reply by cstueckrath** [reply]
or make it possible for clusters to have multiple sites.

**Reply by nomad-cyanide** [reply]
Exactly.

**Reply by nomad-cyanide** [reply]
> @nomad-cyanide @cstueckrath: if you can create a clear series of steps to reproduce the issue with a fresh Netbox 3.2 install, then it can be raised as a ticket.

I will, whenever I have the time to look at 3.2. Currently it doesn't look good. :(
 
> For me: setting a virtual interface on a VM to mode "Access" and then selecting an untagged VLAN works fine; also setting the mode to "Tagged" and selecting a set of tagged VLANs works fine.
> 
> However there are lots of variables. In my case, the VLANs assigned to a particular site, and the VM cluster is assigned to the same site, and I'm not using VLAN groups. So you'll need to be specific around the circumstances required to reproduce this.

No, not a lot of variables.
- A Cluster X with no site assigned and a vm attached to that cluster
- All vlan-groups are either site or site-scopes. No vlan-groups are without site and have vlans defined
- Tagging a vm's in (cluster X) interface with a vlan is not possible in GUI. As it is possible using the API, I believe it is an error in the GUI, at least in version 3.1.9.
